libs/server/Resp/AdminCommands.cs

```diff
@@ -542,13 +543,77 @@ private bool ProcessAdminCommands<TGarnetApi>(RespCommand command, ReadOnlySpan<
             {
                 return ProcessACLCommands(bufSpan, count);
             }
-            else if ((command == RespCommand.REGISTERCS))
+            else if (command == RespCommand.REGISTERCS)
             {
                 return NetworkREGISTERCS(count, recvBufferPtr + readHead, storeWrapper.customCommandManager);
             }
+            else if (command == RespCommand.ASYNC)
+            {
+                if (respProtocolVersion > 2 && count == 1)
+                {
+                    var param = GetCommand(bufSpan, out bool success1);
+                    if (!success1) return false;
+                    if (param.SequenceEqual(CmdStrings.ON) || param.SequenceEqual(CmdStrings.on))
+                    {
+                        useAsync = true;
+                        while (!RespWriteUtils.WriteDirect(CmdStrings.RESP_OK, ref dcurr, dend))
+                            SendAndReset();
+                    }
+                    else if (param.SequenceEqual(CmdStrings.OFF) || param.SequenceEqual(CmdStrings.off))
+                    {
```

Comment:
what should happen here if asyncCompleted < asyncStarted?

Reply:
async is turned off for future requests in this session, existing requests will continue to process and return results asynchronously. if the user wants to prevent this, they need to issue `async barrier` before `async off`.